apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/CoverageResourceProvider.java

```diff
@@ -85,7 +86,9 @@ public Coverage read(@IdParam IdType coverageId) {
       throw new IllegalArgumentException();
 
     Matcher coverageIdMatcher = COVERAGE_ID_PATTERN.matcher(coverageIdText);
-    if (!coverageIdMatcher.matches()) throw new ResourceNotFoundException(coverageId);
+    if (!coverageIdMatcher.matches())
+      throw new IllegalArgumentException("Unsupported ID pattern: " + coverageIdText);
```

Comment:
Does this change make sense to do? Does it have a potential to cause any incompatibility issues?
I believe that the exception here should indicate that the ID parameter pattern is invalid rather than not found. This also enables the ability to more easily test this via the IT tests.

Reply:
NOTE:  The `java.lang.IllegalArgumentException` results in a `ca.uhn.fhir.rest.server.exceptions.InvalidRequestException` exception here as seen in the IT tests. Was following other related code examples, but should these all be changed to `InvalidRequestException`?

Reply:
How do we differentiate between a valid pattern that doesn't have an ID in the database, and an invalid pattern?

Reply:
> How do we differentiate between a valid pattern that doesn't have an ID in the database, and an invalid pattern?
@njdister Further down in the code, a `ResourceNotFoundException` is thrown when the ID is not found. I think your question supports the change for the pattern not matching to throw an `IllegalArgumentException` here instead. Otherwise, would not be able to tell the difference and be able to easily test for this difference.

Reply:
I agree that this change makes sense and is helpful. We shouldn't try to preserve bad exceptions for assumed compatibility.